empty_meta: Add functions that don't depend on Tensor

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #70618
* #70617
* #70616
* __->__ #70615
* #70614
* #70613
* #70612

This adds `at::detail::empty_meta` and
`at::detail::empty_strided_meta` to complement the cpu API.

Differential Revision: [D33623678](https://our.internmc.facebook.com/intern/diff/D33623678)

peterbell10 committed:
empty_meta: Add functions that don't depend on Tensor

[ghstack-poisoned]

peterbell10 committed:
Update on "empty_meta: Add functions that don't depend on Tensor"


This adds `at::detail::empty_meta` and
`at::detail::empty_strided_meta` to complement the cpu API.

[ghstack-poisoned]

peterbell10 committed:
Update on "empty_meta: Add functions that don't depend on Tensor"


This adds `at::detail::empty_meta` and
`at::detail::empty_strided_meta` to complement the cpu API.

[ghstack-poisoned]

peterbell10 committed:
Update on "empty_meta: Add functions that don't depend on Tensor"


This adds `at::detail::empty_meta` and
`at::detail::empty_strided_meta` to complement the cpu API.

[ghstack-poisoned]

peterbell10 committed:
Update on "empty_meta: Add functions that don't depend on Tensor"


This adds `at::detail::empty_meta` and
`at::detail::empty_strided_meta` to complement the cpu API.

[ghstack-poisoned]

peterbell10 committed:
Update on "empty_meta: Add functions that don't depend on Tensor"


This adds `at::detail::empty_meta` and
`at::detail::empty_strided_meta` to complement the cpu API.

[ghstack-poisoned]

peterbell10 committed:
Update on "empty_meta: Add functions that don't depend on Tensor"


This adds `at::detail::empty_meta` and
`at::detail::empty_strided_meta` to complement the cpu API.

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33623678).

peterbell10 committed:
Update on "empty_meta: Add functions that don't depend on Tensor"


This adds `at::detail::empty_meta` and
`at::detail::empty_strided_meta` to complement the cpu API.

Differential Revision: [D33623678](https://our.internmc.facebook.com/intern/diff/D33623678)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33623678).
